server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -104,6 +92,29 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
+    private Insert createSpecificUpdate(VacationUpdate vacationUpdate, Insert baseInsert) {
+        Insert insertWithSubject = addValue(baseInsert, CassandraVacationTable.SUBJECT, vacationUpdate.getSubject());
+        Insert insertWithHtml = addValue(insertWithSubject, CassandraVacationTable.HTML, vacationUpdate.getHtmlBody());
+        Insert insertWithText = addValue(insertWithHtml, CassandraVacationTable.TEXT, vacationUpdate.getTextBody());
+        Insert insertWithFromDate = addValueZonedDateTime(insertWithText, CassandraVacationTable.FROM_DATE, vacationUpdate.getFromDate());
+        Insert insertWithToDate = addValueZonedDateTime(insertWithFromDate, CassandraVacationTable.TO_DATE, vacationUpdate.getToDate());
+        return addValue(insertWithToDate, CassandraVacationTable.IS_ENABLED, vacationUpdate.getIsEnabled());
+    }
+
+    public <T> Insert addValue(Insert insert, String field, MissableOptional<T> missableValue) {
+        if (!missableValue.isMissing()) {
+            return insert.value(field, missableValue.toOptional().orElse(null));
```

Comment:
If you add a `value` method in MissableOptional, you may get rid of the .orElse (which is the responsability of MissableOptional, IMO)